threema-connector-test/src_test/threema/connector/test/process/MultipleRecipientsTest.java

```diff
@@ -0,0 +1,55 @@
+package threema.connector.test.process;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import ch.ivyteam.ivy.bpm.engine.client.BpmClient;
+import ch.ivyteam.ivy.bpm.engine.client.ExecutionResult;
+import ch.ivyteam.ivy.bpm.engine.client.element.BpmElement;
+import ch.ivyteam.ivy.bpm.engine.client.element.BpmProcess;
+import ch.ivyteam.ivy.bpm.exec.client.IvyProcessTest;
+import ch.ivyteam.ivy.environment.AppFixture;
+import ch.ivyteam.ivy.scripting.objects.List;
+import ch.ivyteam.threema.mocks.ThreemaServiceMock;
+import threema.connector.receiverData;
+import threema.connector.sendThreemaMessageData;
+import util.LookupType;
+
+@IvyProcessTest(enableWebServer = true)
+public class MultipleRecipientsTest {
+
+	private final static BpmProcess MULTIPLE_RECIPIENTS_PROCESS = BpmProcess.name("multipleRecipients");
+	private final static String MESSAGE = "Hello World";
+	
+	@BeforeEach
+	void setup(AppFixture fixture) {
+		fixture.config("RestClients.ThreemaGateway.Url", ThreemaServiceMock.URI);
+	}
+	
+	
+	@Test
+	void prepareMultipleRecipients(BpmClient bpmClient) {
+		List<String> recipients = new List<String>();
+		recipients.add("invalidThreemaID");
+		recipients.add("41000000000");
+		recipients.add("invalid@email.ch");
+		
+		BpmElement callable = MULTIPLE_RECIPIENTS_PROCESS.elementName("call(String,List<String>)");	
+		ExecutionResult result = bpmClient.start().subProcess(callable).execute(MESSAGE, recipients);
+		sendThreemaMessageData msgData = result.data().last();
+		List<receiverData> listReceiver = msgData.getReceiverData();
+		
+		assertThat(listReceiver.size()).isEqualTo(recipients.size());
+		assertThat(listReceiver.get(0).getType()).isEqualTo(LookupType.THREEMAID);
+		assertThat(listReceiver.get(1).getType()).isEqualTo(LookupType.PHONE);
+		assertThat(listReceiver.get(2).getType()).isEqualTo(LookupType.EMAIL);
```

Comment:
you could simplify this:
 ```
assertThat(listReceiver).extracting(receiverData::getType)
  .containsExactly(LookupType.THREEMAID,LookupType.PHONE, LookupType.EMAIL );
```
in case of an error, you get much more details as you will see the complete list at once compared with the expected values.

Reply:
implemented suggested change